pyproject.toml

```
@@ -1,10 +1,181 @@
 [build-system]
-requires = ["setuptools", "wheel"]
-build-backend = "setuptools.build_meta"
+requires = ["hatchling"]
+build-backend = "hatchling.build"
+
+[tool.hatch.publish.index]
+disable = true  # Requires confirmation when publishing to pypi.
+
+[project]
+name = "s2spy"
+dynamic = ["version"]
+description = "python package for s2s forecasts with ai"
+readme = "README.md"
+license = "Apache-2.0"
+requires-python = ">=3.8,<3.11"
+authors = [
+  {email = "y.liu@esciencecenter.nl"},
+  {name = "Yang Liu, Bart Schilperoort, Peter Kalverla, Jannes van Ingen, Sem Vijverberg"}
+]
+maintainers = [
+  {name = "Yang Liu", email = "y.liu@esciencecenter.nl"},
+  {name = "Bart Schilperoort", email = "b.schilperoort@esciencecenter.nl"},
+  {name = "Peter Kalverla", email = "b.schilperoort@esciencecenter.nl"},
+  {name = "Jannes van Ingen", email = "jannes.van.ingen@s2s-ai.com"},
+  {name = "Sem Vijverberg", email = "sem.vijverberg@vu.nl"},
+]
+keywords = [
+    "AI",
+    "S2S",
+]
+classifiers = [
+    "Development Status :: 2 - Pre-Alpha",
+    "Intended Audience :: Developers",
+    "License :: OSI Approved :: Apache Software License",
+    "Natural Language :: English",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+]
+dependencies = [
+    "lilio",
+    "matplotlib",
+    "netcdf4",
+    "numpy",
+    "pandas",
+    "scikit-learn",
+    "scipy",
+    "xarray",
+]
+
+[project.optional-dependencies]
+dev = [
+    "bump2version",
+    "hatch",
+    "ruff",
+    "black",
+    "isort",
+    "mypy",
+    "pytest",
+    "pytest-cov",
+]
+
+docs = [  # Required for ReadTheDocs
+  "myst_parser",
+  "sphinx",
+  "sphinx_rtd_theme",
+  "sphinx-autoapi",
+  "coverage[toml]",
+]
+
+[project.urls]
+"Bug Tracker" = "https://github.com/AI4S2S/ai4s2s/issues"
+Documentation = "https://ai4s2s.readthedocs.io/"
+Homepage = "https://github.com/AI4S2S/ai4s2s"
+
+[tool.hatch.version]
+path = "s2spy/__init__.py"
+
+[tool.hatch.build.targets.sdist]
+exclude = [
+  "/.github", "/.mypy_cache", "/.pytest_cache", "/.githooks",
+  "sonar-project.properties"
+]
+
+[tool.hatch.build.targets.wheel]
+packages = ["s2spy"]
+
+[tool.hatch.envs.default]
+features = ["dev"]
+
+[tool.hatch.envs.default.scripts]
+lint = [
+  "ruff check s2spy",
+  "mypy s2spy",
```

Why not lint and type check the tests as well?

```suggestion
  "ruff check .",
  "mypy .",
```

And then you can ignore the docstring errors for the test folder like this (see lilio for reference):
```toml
[tool.ruff.per-file-ignores]
"tests/**" = ["D"]
```